modin/experimental/xgboost/test/test_xgboost.py

```diff
@@ -0,0 +1,300 @@
+    modin_X = pd.DataFrame(X)
+    modin_y = type_modin_y(y)
+    dmatrix_mxgb = xgb.DMatrix(modin_X, label=modin_y)
+
+    evals_result_xgb = {}
+    evals_result_mxgb = {}
+    bst = xgboost.train(
+        param,
+        dmatrix_xgb,
+        num_round,
+        evals_result=evals_result_xgb,
+        evals=[(dmatrix_xgb, "train")],
+    )
+    modin_bst = xgb.train(
+        param,
+        dmatrix_mxgb,
+        num_round,
+        evals_result=evals_result_mxgb,
+        evals=[(dmatrix_mxgb, "train")],
+        num_actors=num_actors,
+    )
+
+    predictions = bst.predict(dmatrix_xgb)
+    modin_predictions = modin_bst.predict(dmatrix_mxgb, num_actors=num_actors)
+
+    best_preds = np.asarray([np.argmax(line) for line in predictions])
+    modin_best_preds_ = np.asarray(
+        [np.argmax(line) for line in modin_predictions.to_numpy()]
+    )
+    modin_best_preds = pd.DataFrame(modin_best_preds_)
+
+    preds = pd.DataFrame(best_preds).apply(lambda x: round(x))
+    modin_preds = modin_best_preds.apply(lambda x: round(x))
+
+    val = accuracy_score(y, preds)
+    modin_val = accuracy_score(modin_y, modin_preds)
+
+    np.testing.assert_allclose(val, modin_val)
+
+
+@pytest.mark.parametrize(
+    "type_modin_y",
+    [pd.DataFrame, pd.Series],
+)
+@pytest.mark.parametrize(
+    "num_round",
+    [10, 100, 300],
+)
+@pytest.mark.parametrize(
+    "num_actors",
+    [1, num_cpus, None, 400],
+)
+@pytest.mark.parametrize(
+    "data",
+    [
+        (
+            load_diabetes(),
+            {
+                "n_estimators": 500,
+                "max_depth": 4,
+                "min_samples_split": 5,
+                "learning_rate": 0.01,
+                "loss": "ls",
+            },
+        ),
+    ],
+    ids=[
+        "load_diabetes",
+    ],
+)
+def test_dmatrix_train_predict_regression_datasets(
+    data, num_actors, num_round, type_modin_y
+):
+    dataset, param = data
+
+    X = dataset.data
+    y = dataset.target
+    dmatrix_xgb = xgboost.DMatrix(X, label=y)
+    with pytest.raises(AssertionError):
+        dmatrix_mxgb = xgb.DMatrix(X, label=y)
+
+    modin_X = pd.DataFrame(X)
+    modin_y = type_modin_y(y)
+    dmatrix_mxgb = xgb.DMatrix(modin_X, label=modin_y)
+
+    evals_result_xgb = {}
+    evals_result_mxgb = {}
+    bst = xgboost.train(
+        param,
+        dmatrix_xgb,
+        num_round,
+        evals_result=evals_result_xgb,
+        evals=[(dmatrix_xgb, "train")],
+    )
+    modin_bst = xgb.train(
+        param,
+        dmatrix_mxgb,
+        num_round,
+        evals_result=evals_result_mxgb,
+        evals=[(dmatrix_mxgb, "train")],
+        num_actors=num_actors,
+    )
+
+    predictions = bst.predict(dmatrix_xgb)
+    modin_predictions = modin_bst.predict(dmatrix_mxgb, num_actors=num_actors)
+
+    best_preds = predictions
+    modin_best_preds = modin_predictions
+
+    preds = pd.DataFrame(best_preds).apply(lambda x: round(x))
+    modin_preds = modin_best_preds.apply(lambda x: round(x))
```

Comment:
As I described in previous comments, this is not correct way to check accuracy of regression models. To check the regression accuracy you can use `sklearn.metrics.mean_squared_error`, for example:
```suggestion
    import from sklearn.metrics.mean_squared_error # move to the top of file
    preds = mean_squared_error(y, best_preds)
    modin_preds = mean_squared_error(y, modin_best_preds)
```

In this case you will have MSE metric. The difference of native/modin is <0.1%

Reply:
`accuracy_check` was changed on `mean_square_error`